packages/grafana-data/src/types/fieldOverrides.ts

```diff
@@ -54,16 +54,16 @@ export interface FieldOverrideEditorProps<TValue, TSettings> extends Omit<Standa
   context: FieldOverrideContext;
 }
 
-export interface FieldConfigEditorConfig<TSettings = any> {
-  id: string;
+export interface FieldConfigEditorConfig<TOptions, TSettings = any> {
+  id: (keyof TOptions & string) | string;
   name: string;
   description: string;
   settings?: TSettings;
   shouldApply?: (field: Field) => boolean;
```

Comment:
can you add comment here?  I remember adding this function, but do not remember why!

Reply:
yes!

Reply:
what is it for, is to check that the value is of right type before passing it to the processor? but can't it be handled inside the processor?